### DISCUSSION TITLE
Is there any option to re-arrange the elements before return

### DISCUSSION THREAD

**Discussion by RameshKrish29** [remarkablemark/html-react-parser#920] [General]
I was getting html content from API, that data was managed by backend(Admin Panel). If anyone enter the order wrongly. I should re-arrange the order and display the content in right order.

For example: 

The right format of the content is below

_<section>
    <em>first title with italic style</em>
    <span>second title with normal style</span>
    <strong>third title with bold style</strong>
</section>_

If they entered wrongly, I have displayed the content like below
![image](https://user-images.githubusercontent.com/91420119/235156344-68bb233f-7cf8-4d5f-aa2e-92028000b001.png)

I want the method to re-arrange the order of the content before rendering. Is there any methods or example? please help me on this.

**Comment by remarkablemark** [comment]
@RameshKrish29 you can use the [`replace`](https://github.com/remarkablemark/html-react-parser#replace) option.

Let's say you have the HTML string:

```html
<ol>
  <li>first title with italic style</li>
  <li>second title with normal style</li>
  <li>third title with bold style</li>
</ol>
```

Let's save the value in variable `html`:

```tsx
const html = `
  <ol>
    <li>first title with italic style</li>
    <li>second title with normal style</li>
    <li>third title with bold style</li>
  </ol>
`;
```

To reverse the order of the children inside the `ol`:

```tsx
import parse, { domToReact, Element, DOMNode } from 'html-react-parser';

parse(html, {
  replace: (domNode: DOMNode) => {
    if (domNode instanceof Element && domNode.name === 'ol') {
      return <>{domToReact(domNode.children.reverse())}</>;
    }
  }
});
```

This means you can reorder the DOM nodes before using [`domToReact`](https://github.com/remarkablemark/html-react-parser#replace-element-and-children) to render the replaced elements.

See [CodeSandbox](https://codesandbox.io/s/html-react-parser-920-2v7wl8?file=/src/index.tsx).
